src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -474,15 +487,27 @@ private void validateIntegerPartitioningColumn(String columnName, FailureCollect
     }
   }
 
-  private void validateTimePartitioningColumn(String columnName, FailureCollector collector, Schema fieldSchema) {
+  private void validateTimePartitioningColumn(String columnName, FailureCollector collector,
+                                              Schema fieldSchema, TimePartitioning.Type timePartitioningType) {
+
+    // If timePartitioningType is HOUR, then logicalType cannot be DATE
     Schema.LogicalType logicalType = fieldSchema.getLogicalType();
-    if (logicalType != LogicalType.DATE && logicalType != LogicalType.TIMESTAMP_MICROS
-      && logicalType != LogicalType.TIMESTAMP_MILLIS) {
-      collector.addFailure(
-        String.format("Partition column '%s' is of invalid type '%s'.", columnName, fieldSchema.getDisplayName()),
-        "Partition column must be a date or timestamp.")
-        .withConfigProperty(NAME_PARTITION_BY_FIELD)
-        .withOutputSchemaField(columnName).withInputSchemaField(columnName);
+
+    if (timePartitioningType == TimePartitioning.Type.HOUR) {
```

Comment:
Add some comments for the check.

Reply:
// If timePartitioningType is HOUR, then logicalType cannot be DATE Only TIMESTAMP_MICROS and TIMESTAMP_MILLIS


    // For any other timePartitioningType (DAY, MONTH, YEAR), logicalType can only be TIMESTAMP_MICROS, TIMESTAMP_MILLIS